README.rst

```diff
@@ -110,147 +145,237 @@ version of the AWS Encryption SDK, we recommend using the default value.
+    from aws_cryptographic_materialproviders.mpl.config import MaterialProvidersConfig
+    ffrom aws_cryptographic_materialproviders.mpl.models import (
+        CreateAwsKmsDiscoveryKeyringInput,
+        DiscoveryFilter,
+    )
+    from aws_cryptographic_materialproviders.mpl.references import IKeyring
+
     import aws_encryption_sdk
-    from aws_encryption_sdk.key_providers.kms import DiscoveryFilter
+    from aws_encryption_sdk import CommitmentPolicy
 
-    discovery_filter = DiscoveryFilter(
-        account_ids=['222222222222', '333333333333'],
-        partition='aws'
+    # Instantiate the encryption SDK client.
+    client = aws_encryption_sdk.EncryptionSDKClient(
+        commitment_policy=CommitmentPolicy.REQUIRE_ENCRYPT_REQUIRE_DECRYPT
     )
-    kms_key_provider = aws_encryption_sdk.DiscoveryAwsKmsMasterKeyProvider(
-        discovery_filter=discovery_filter
+
+    # Create a Discovery keyring to use for decryption
+    mat_prov: AwsCryptographicMaterialProviders = AwsCryptographicMaterialProviders(
+        config=MaterialProvidersConfig()
+    )
+
+    discovery_keyring_input: CreateAwsKmsDiscoveryKeyringInput = CreateAwsKmsDiscoveryKeyringInput(
+        kms_client=boto3.client('kms', region_name="us-east-1"),
+        discovery_filter=DiscoveryFilter(
+            account_ids=["2222222222222"],
+            partition="aws"
+        )
     )
 
+    discovery_keyring: IKeyring = mat_prov.create_aws_kms_discovery_keyring(
+        input=discovery_keyring_input
+    )
+
+
 If you do not want to filter the set of allowed accounts, you can also omit the ``discovery_filter`` argument.
 
-Note that a ``DiscoveryAwsKmsMasterKeyProvider`` cannot be used for encryption operations.
+Note that an ``AwsKmsDiscoveryKeyring`` cannot be used for encryption operations.
 
 Encryption and Decryption
 =========================
-After you create an instance of an ``EncryptionSDKClient`` and a ``MasterKeyProvider``, you can use either of
+After you create an instance of an ``EncryptionSDKClient`` and a ``Keyring``, you can use either of
 the client's two ``encrypt``/``decrypt`` functions to encrypt and decrypt your data.
 
-.. code:: python
-
-    import aws_encryption_sdk
-    from aws_encryption_sdk.identifiers import CommitmentPolicy
-
-    client = aws_encryption_sdk.EncryptionSDKClient(
-        commitment_policy=CommitmentPolicy.FORBID_ENCRYPT_ALLOW_DECRYPT
-    )
+Here's an example for using a KMS keyring for encryption and decryption:
 
-    kms_key_provider = aws_encryption_sdk.StrictAwsKmsMasterKeyProvider(key_ids=[
-        'arn:aws:kms:us-east-1:2222222222222:key/22222222-2222-2222-2222-222222222222',
-        'arn:aws:kms:us-east-1:3333333333333:key/33333333-3333-3333-3333-333333333333'
-    ])
-    my_plaintext = b'This is some super secret data!  Yup, sure is!'
+.. code:: python
 
-    my_ciphertext, encryptor_header = client.encrypt(
+    # Encrypt the data.
```

Comment:
Can we make this more 1:1 with what was left here? i.e. create a client, create an AwsKmsMultiKeyring, encrypt/decrypt, asserts